File: core/tmp/Java/src/com/dbc/algorithm/ReferOffer/CongWeiDaoTouDaYinLianBiaoLcof.java
package com.dbc.algorithm.ReferOffer;

import java.util.Stack;

public class CongWeiDaoTouDaYinLianBiaoLcof {
    public class ListNode {
        int val;
        ListNode next;

        ListNode(int x) {
            val = x;
        }
    }

    public int[] reversePrint(ListNode head) {
        Stack<Integer> stack = new Stack<>();
        while (head != null) {
            stack.push(head.val);
            head = head.next;
        }

        int[] result = new int[stack.size()];
        int lenght = stack.size();
        for (int i = 0; i < lenght; i++) {
            result[i] = stack.pop();
        }
        return result;
    }
}

File: core/tmp/Python/refer-offer/cong_wei_dao_tou_da_yin_lian_biao_lcof.py
from typing import *

class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None

def reversePrint(head: ListNode) -> List[int]:
    stack = list()
    result = []
    while head is not None:
        stack.append(head.val)
        head = head.next
    
    while len(stack) != 0:
        result.append(stack.pop())
    
    return result